grokking-algorithms/chapter-one

```diff
@@ -0,0 +1,58 @@
+# Grokking Algorthims Chapter one
+
+"An algorith is a set of instructions for accomplishing a task"
+
+Trade offs - performance/ data structure/ which algorithm is used/ scalability/ running time/ memory
+
+
+## Binary search
+
+[comparison with linear search](https://jorgechavez.dev/2020/08/22/everything-you-need-to-know-about-binary-search-algorithm/)
+
+
+```python
+def binary_search (list, item):
+    high = len(list) - 1
+     low = 0
+     while low <= high:
+        pivot = (high + low ) // 2
+        if list[pivot] == item:
+            return pivot
+        if list[pivot] > item:
+            high = pivot - 1
+        if list[pivot] < item:
+            low = pivot + 1
+    return None 
+
+
```

Comment:
```suggestion
def binary_search(input, item):
    high = len(list) - 1
    low = 0
    while low <= high:
        pivot = (high + low) // 2
        if input[pivot] == item:
            return pivot
        if input[pivot] > item:
            high = pivot - 1
        if input[pivot] < item:
            low = pivot + 1
    return
```

because list is a reserved word, and return returns None if nothing else is specified.

Reply:
<3 our python queen!